tools/c7n_mailer/c7n_mailer/slack_delivery.py

```diff
@@ -123,7 +123,8 @@ def retrieve_user_im(self, email_addresses):
                 headers={'Content-Type': 'application/x-www-form-urlencoded',
                          'Authorization': 'Bearer %s' % self.config.get('slack_token')}).json()
 
-            if not response["ok"] and "Retry-After" in response["headers"]:
+            if not response["ok"] and "headers" in response.keys() and \
```

Comment:
if you don't mind, could you pull the `if not response["ok"]` into a parent block to prevent the copy/paste in lines 133 and 135?